### ISSUE TITLE
Tried to add health diff to missing part BodyPartRecord(Eye parts.Count=0)

### ISSUE THREAD

**Issue by ANRimworld** [juanosarg/VanillaGeneticsExpanded#4]
Hi :) I came across this while trying to add a "Humanoid Implant Compound Insectoid Eye" to a pawn with a missing right eye.

Was able to recreate the issue without fail. In comparing Recipe_InstallGeneticBodyPart to Rimworld's Recipe_InstallArtificialBodyPart; I noticed you aren't using MedicalRecipesUtility.RestorePartAndSpawnAllPreviousParts.

I'm not sure if this is intended, I don't see anything that'd imply the implants are meant to augment only healthy parts. But it does error and consume your implant either way.

I added the following code to your Apply on pawn, and it resolved the issue and I didn't notice any unintended effects. Though will update if I find anything while I play more.
```
TaleRecorder.RecordTale(TaleDefOf.DidSurgery, billDoer, pawn); //Add starts here
if (!pawn.health.hediffSet.GetNotMissingParts(BodyPartHeight.Undefined, BodyPartDepth.Undefined, null, null).Contains(part))//Making it only happen when missing part. Unsure if necessary 				
{
MedicalRecipesUtility.RestorePartAndSpawnAllPreviousParts(pawn, part, billDoer.Position, billDoer.Map); //Added to see if it fixes issue with missing part
}

```

**Comment by juanosarg** [comment]
I added the custom class because without it you could get infinite Hearts.... Could you test if that issue still happens with what you added? Basically adding a heart genetic implant to a pawn with a healthy heart would produce a Heart implant. Then you remove the genetic implant and install it again => endless hearts

**Comment by ANRimworld** [comment]
Seems like that's not an issue with this. I suspect because I added the check to only run that utility if the part is missing. (I guessed you didnt remove it for no good reason) So if there's a healthy heart it wont try to spawn a previous part.

My Test:
Install chemfuel heart: Nothing popped out (no heart)
Remove Chemfuel heart: Implant popped out - Pawn still had a regular heart

**Comment by ANRimworld** [comment]
Actually thinking about it, there's may not be a point in using RestorePartAndSpawnAllPreviousParts at all. 
Just pawn.health.RestorePart(part, null, true); with the same check on if the part is missing would probably do the trick.

**Comment by juanosarg** [comment]
Could you please do this in a pull request? I'm having some seriously busy days at work

**Comment by ANRimworld** [comment]
Sure, I was about to create it, when I decided to test one more thing, and there is a way to create infinite parts with this. I've identified why, but unsure the best fix. 

From what I can tell, it comes down to Remove Part Vs Remove Implant. You have created recipes for remove implant on all of the implants. Which you then harmony to get your quality back.

The issue, is from what I can tell is remove implant is not meant for a "part" eg eye/ear etc. But for things like joywire or your neuron reinforcement.

But for parts it creates 2 recipes that do similar things. 
Remove part is basegame, this handles the correct removing of the part leaving no eye. BUT it kills the quality setting it to normal. 
![image](https://user-images.githubusercontent.com/107319432/173248538-8850c596-add0-4adb-86b8-c6a3e58786b5.png)

Your Recipe: Handles the quality correctly. BUT it will leave the pawn with a healthy eye. (Which could then be harvested, then install implant again repeat)
![image](https://user-images.githubusercontent.com/107319432/173248613-1a827f43-e9aa-4b79-a3b4-878b60b4a25e.png)

So based on this, I believe the fix would be 2 things. 

- Remove Recipes for "parts" from your recipedefs and have it handled by the games Remove part
- Add a similar harmony patch onto Recipe_RemoveBodyPart to handle the quality

Let me know what you think/If there's issues you are aware of in this solution. I could possibly make the changes, but its been a minute since I've worked in the harmony/reflection world. 

Also I could still create the pull for the original fix. This issue is the lesser of 2 evils in my mind.
